src/azul/plugins/metadata/hca/transform.py

```diff
@@ -1070,50 +1071,53 @@ def transform(self) -> Iterable[Contribution]:
                                 ),
                                 projects=[self._project(project)])
                 # Supplementary file matrices provide stratification values that
-                # we need to add to the contents of the contribution.
+                # need to be reflected by inner entities in the contribution.
                 if isinstance(file, api.SupplementaryFile):
-                    if self._is_contributor_matrix(file) or self._is_dcp2_matrix(file):
-                        self.add_matrix_stratification_to_contents(contents, file)
+                    if Submitter.category_for(file) is not None:
+                        contents = {
+                            **contents,
+                            **self.matrix_stratification_values(file)
+                        }
                 yield self._contribution(contents, file.document_id)
 
-    def add_matrix_stratification_to_contents(self,
-                                              contents: JSON,
-                                              file: api.SupplementaryFile) -> None:
-        """
-        Inject stratification values from a CGM supplementary file into the
-        inner entities of the file contribution document.
-        """
+    def matrix_stratification_values(self,
+                                     file: api.SupplementaryFile) -> JSON:
         points = {
             dimension: values.split(',')
             for dimension, values in (point.split('=')
                                       for stratum in file.json['file_description'].split('\n')
                                       for point in stratum.split(';'))
         }
-        donor = {
-            **({'genus_species': sorted(points['genusSpecies'])}
-               if 'genusSpecies' in points else {}),
-            **({'development_stage': sorted(points['developmentStage'])}
-               if 'developmentStage' in points else {}),
-        }
-        if donor:
-            donor['biomaterial_id'] = uuid.uuid4()
-            contents['donors'].append(donor)
-        if organ := points.get('organ'):
+        contents = {}
+        donors = {}
+        genus_species = points.get('genusSpecies')
+        if isinstance(genus_species, list):
```

Comment:
This looks convoluted.  AFAICT, genus_species is either None or a non-empty list. This looks as if you are expecting it to be other types. Why can't you use `is not None`?